qiskit_optimization/problems/linear_constraint.py

```diff
@@ -79,3 +79,17 @@ def evaluate(self, x: Union[ndarray, List, Dict[Union[int, str], float]]) -> flo
             The left-hand-side of the constraint given the variable values.
         """
         return self.linear.evaluate(x)
+
+    def __repr__(self):
+        # pylint: disable=cyclic-import
+        from ..translators.prettyprint import _expr2str, DEFAULT_TRUNCATE
```

Comment:
Alright. I rename `_expr2str` into `expr2str`

It's hard to avoid cyclic import between prettyprint and problem objects because prettyprint needs to access these problem objects.

Reply:
Ok, I figured it was mots likely a non-trivial task to deal with the cyclic import.